cli-files/go-cat/cmd/root_test.go

```diff
@@ -0,0 +1,48 @@
+package cmd
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+func TestRootCmd(t *testing.T) {
+	t.Run("No path provided", func(t *testing.T) {
+
+		cmd := rootCmd
+		err := cmd.Execute()
+
+		if err == nil {
+			t.Error("Expected error, got nil")
+		}
+	})
+
+	t.Run("path to file dew.txt", func(t *testing.T) {
+		r, w, _ := os.Pipe()
+		os.Stdout = w
```

Comment:
Redirecting the _actual_ `os.Stdout`/`os.Stderr` can be confusing, e.g. because the test runner may try to write things there (and also, this may leak into other tests). Fortunately, cobra has tooling for redirecting stdout which you can use to avoid this: https://gianarb.it/blog/golang-mockmania-cli-command-with-cobra